doc/modules/sgd.rst

```diff
@@ -561,12 +561,12 @@ The code is written in Cython.
       <1107.2490v2>`
       Xu, Wei (2011)
 
-   .. [#5] `"Regularization and variable selection via the elastic net"
-      <http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.124.4696>`_
+   .. [#5] :doi:`"Regularization and variable selection via the elastic net"
       H. Zou, T. Hastie - Journal of the Royal Statistical Society Series B,
       67 (2), 301-320.
+      <https://www.jstor.org/stable/3647580>`
```

Comment:
The DOI here is
[10.1111/j.1467-9868.2005.00503.x](https://doi.org/10.1111/j.1467-9868.2005.00503.x)